### DISCUSSION TITLE
Dataset available?

### DISCUSSION THREAD

**Discussion by calebrob6** [Clay-foundation/model#146] [Q&A]
Hey all, is the Clay dataset available? We'd like to pre-train a location encoder on it if possible.  

cc @konstantinklemmer

**Comment by brunosan** [comment]
You mean the input dataset? The script to create it is [this one](https://github.com/Clay-foundation/model/tree/main/scripts). Not sure it's valuable as a ready-made dataset, but if you want access to the bucket, happy to give you (and others) access.

If you mean the embeddings for the input data, they are here https://beta.source.coop/repositories/clay/clay-model-v0-embeddings/description/

**Comment by konstantinklemmer** [comment]
Actually that would be great - I suppose the dataset / dataloaders in here will work with the data stored in the bucket?
[https://github.com/Clay-foundation/model/blob/main/src/datamodule.py](https://github.com/Clay-foundation/model/blob/main/src/datamodule.py)

**Reply by lauracchen** [reply]
Hi @konstantinklemmer happy to share access to the S3 bucket of training data for Clay v0. Can you shoot me an email to coordinate access - laura@madewithclay.org? Thanks!

**Reply by konstantinklemmer** [reply]
Hi @lauracchen! Thanks, I already got the access details from @calebrob6.

**Comment by konstantinklemmer** [comment]
Hi @brunosan! Quick question for my understanding. I run the following code to get a Clay batch:

```
loader = GeoTIFFDataPipeModule(data_dir="s3://clay-tiles-02", batch_size=32, num_workers=8)
loader.setup(stage="fit")

#Get first batch of data from train dataloader
for batch in loader.train_dataloader():
    print(batch)
    break
```

When I get the shape of the training images `batch['image'].shape` I get `torch.Size([32, 13, 512, 512])`. 

Why do I get only 13 channels? Should this not be a stack of Sentinel-1, Sentinel-2 and the DEM? 

Sorry if this is a stupid question, but I couldn't find the answer myself.

**Reply by lauracchen** [reply]
Hi @konstantinklemmer thanks for checking in! 13 channels is correct! This is a stack of 10 bands of Sentinel-2, 2 bands of Sentinel-1, and 1 band of DEM data. You can find more on the model release notes [here](https://clay-foundation.github.io/model/specification.html).

**Reply by konstantinklemmer** [reply]
Got it, thank you!

**Comment by konstantinklemmer** [comment]
Another quick question @lauracchen: You have made the embeddings for all training data available very conveniently here: https://beta.source.coop/repositories/clay/clay-model-v0-embeddings/description/.

Are you also hosting the val and test set embeddings on the same platform or somewhere else? Thanks for all your help throughout!

**Reply by weiji14** [reply]
Those embeddings are generated from all of the GeoTIFF files we had (which includes both training + validation, there is no test set). Do you need separate train/validation embeddings for some reason? If so, we would need to capture the indexes from the code here https://github.com/Clay-foundation/model/blob/3f4210ff3160240e73c4d3541962032afef957db/src/datamodule.py#L304-L309

**Reply by konstantinklemmer** [reply]
Got it! No, that makes sense. I was just confused by the wording on the website the embeddings are stored on.
![image](https://github.com/Clay-foundation/model/assets/26688151/4a87aa5e-3f5b-43a7-864f-24ad8c817f87)
